src/domain/synthetics/markets/utils.ts

```diff
@@ -162,6 +176,31 @@ export function getAvailableUsdLiquidityForCollateral(marketInfo: MarketInfo, is
   return liqudiity;
 }
 
+export function getMarketReservesAccountingInterest(marketInfo: MarketInfo, isLong: boolean) {
+  if (marketInfo.isSpotOnly) {
+    return {
+      reserveUsd: BigNumber.from(0),
+      maxReserveUsd: BigNumber.from(0),
+    };
+  }
+
+  const reservedUsd = getReservedUsd(marketInfo, isLong);
+  const maxReservedUsd = getMaxReservedUsd(marketInfo, isLong);
+
+  const openInterestUsd = getOpenInterestUsd(marketInfo, isLong);
+  const maxOpenInterestUsd = getMaxOpenInterestUsd(marketInfo, isLong);
+
+  const [currentMaxReserve, currentReserve] = getArrayWithSmallerDifference(
```

Comment:
`getArrayWithSmallerDifference` is used only in this specific place, i think it's simpler to keep calculation logic here instead of moving to separate function?